src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2196,6 +2224,25 @@ export class SavedObjectsRepository {
       );
     }
   }
+
+  /** Validate a migrated doc against the registered saved object type's schema. */
+  private validateObjectAttributes<T>(type: string, doc: SavedObjectSanitizedDoc<T>) {
+    const savedObjectType = this._registry.getType(type);
+    if (!savedObjectType?.schemas) {
+      return;
+    }
+
+    const validator = new SavedObjectsTypeValidator({
+      type,
+      validationMap: savedObjectType.schemas,
+    });
+
+    try {
+      validator.validate(this._migrator.kibanaVersion, doc.attributes);
+    } catch (error) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(error.message);
```

Comment:
We decorate bad requests with the reason provided, so this should be ok as an initial implementation. The `error.message` here will be of the form `The ${kibanaVersion} validation for saved object of type [${this.type}] is malformed.` and will surface in logs.